PR: https://github.com/odoo/odoo/pull/215789

From: edc6562db9a8eff0bf5b92bbda17d36af30927f3
From: alan-odoo
Date: 2025-09-06 21:40:47

Breaking data model changes scores: del:4 + add:0, change matches:
-    _inherit = 'res.users'
-    equipment_ids = fields.One2many('maintenance.equipment', 'owner_user_id', string="Managed Equipment")
-    equipment_count = fields.Integer(related='employee_id.equipment_count', string="Assigned Equipment")

Total Changes: 39

[IMP] *: update the res.users form views

This commit modifies the fields on the preferences / profile form, which
allows users to read or modify their own data depending on their
permissions, and on the user form for the administrators. The aim of this
change is to display more relevant information to users.

The hr.res_users_view_form_simple_modif view has been removed and merged in
the mail_bot_hr.res_users_view_form_simple_modif view since it must not be
used anymore for the preferences form as this one must not have a different
display when the hr module is installed.

As some fields are removed from the forms and are not longer used, they are
also removed from the res.users model.

Enterprise PR: https://github.com/odoo/enterprise/pull/88518
Upgrade PR: https://github.com/odoo/upgrade/pull/8207

Task-4775990

Part-of: odoo/odoo#215789
Related: odoo/enterprise#88518
Related: odoo/upgrade#8207
Signed-off-by: Thibault Delavallee (tde) <tde@openerp.com>

================================= pseudo patch: =================================

--- a/addons/hr_maintenance/models/__init__.py
+++ b/addons/hr_maintenance/models/__init__.py
@@ -1,5 +1,5 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
 from . import equipment
-from . import res_users
+from . import hr_employee
 from . import hr_employee_public

--- a/addons/hr_maintenance/models/res_users.py
+++ b/addons/hr_maintenance/models/hr_employee.py
@@ -1,17 +1,6 @@
 from odoo import api, models, fields
 
 
-class ResUsers(models.Model):
-    _inherit = 'res.users'
-
-    equipment_ids = fields.One2many('maintenance.equipment', 'owner_user_id', string="Managed Equipment")
-    equipment_count = fields.Integer(related='employee_id.equipment_count', string="Assigned Equipment")
-
-    @property
-    def SELF_READABLE_FIELDS(self):
-        return super().SELF_READABLE_FIELDS + ['equipment_count']
-
-
 class HrEmployee(models.Model):
     _inherit = 'hr.employee'
 
